website/docs/r/lb_app_rule.html.markdown

```diff
@@ -0,0 +1,104 @@
+---
+layout: "vcd"
+page_title: "vCloudDirector: vcd_lb_app_rule"
+sidebar_current: "docs-vcd-resource-lb-app-rule"
+description: |-
+  Provides an NSX edge gateway load balancer application rule resource.
+---
+
+# vcd\_lb\_app\_rule
+
+Provides a vCloud Director Edge Gateway Load Balancer Application Rule resource. An application rule
+allows to directly manipulate and manage IP application traffic with load balancer.
+
+~> **Note:** To make load balancing work one must ensure that load balancing is enabled on edge
+gateway (edge gateway must be advanced).
+This depends on NSX version to work properly. Please refer to [VMware Product Interoperability
+Matrices](https://www.vmware.com/resources/compatibility/sim/interop_matrix.php#interop&29=&93=) 
+to check supported vCloud director and NSX for vSphere configurations.
+
+~> **Note:** The vCloud Director API for NSX supports a subset of the operations and objects defined
+in the NSX vSphere API Guide. The API supports NSX 6.2, 6.3, and 6.4.
+
+Supported in provider *v2.4+*
+
+## Example Usage 1 (Application rule with single line script)
+
+```hcl
+provider "vcd" {
+  user     = "${var.admin_user}"
+  password = "${var.admin_password}"
+  org      = "System"
+  url      = "https://AcmeVcd/api"
+}
+
+resource "vcd_lb_app_rule" "example-one" {
+  edge_gateway = "my-edge-gw"
+  org          = "my-org"
+  vdc          = "my-org-vdc"
+
+  name = "script1"
+  script = [
+    "acl vmware_page url_beg / vmware redirect location https://www.vmware.com/ ifvmware_page",
+  ]
+}
+```
+
+## Example Usage 2 (Application rule with multi line script)
+
+```hcl
+provider "vcd" {
+  user     = "${var.admin_user}"
+  password = "${var.admin_password}"
+  org      = "System"
+  url      = "https://AcmeVcd/api"
+}
+
+resource "vcd_lb_app_rule" "example-two" {
+  edge_gateway = "my-edge-gw"
+  org          = "my-org"
+  vdc          = "my-org-vdc"
+  name         = "script1"
+  script = [
+    "acl vmware_page url_beg / vmware redirect location https://www.vmware.com/ ifvmware_page",
+    "acl other_page2 url_beg / other2 redirect location https://www.other2.com/ ifother_page2",
+    "acl hello payload(0,6) -m bin 48656c6c6f0a"
```

Comment:
Interesting approach to divide lines. Why such? Does this come from below API? We have one place where we define scripts - the `initscript` - is its way of any use here?

Reply:
I might have overengineered with this. I'll have a look if Terraform could accept raw string (with naturally including newlines) and what `initscript` does.

Reply:
Here's another take at it. Apparently Terraform has a HEREDOC syntax which plays nicely here.

Reply:
Fantastic!